packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -149,6 +149,21 @@ class ShaderMask extends SingleChildRenderObjectWidget {
 ///
 /// This effect is relatively expensive, especially if the filter is non-local,
 /// such as a blur.
+///
+/// The effect is applied to all existing painted content. To constrain
```

Comment:
that seems like a bug, if so. I would have expected it to create a new layer, the size of the render object, then apply the effect to everything in the part of the parent layer that is directly under that layer.